### PR TITLE
Enable todor() to produce nice markdown output

### LINKED ISSUE #38: It is possible to capture the output into an object or file?
I am thinking of generating a nice current todo list that can be embedded in my readme file for instance.

### PR DESCRIPTION
**Pull request by mcguinlu**
(Partially) fixes #38.

Hi @dokato, I implemented a fix for this issue for my own use, so said I would open a PR in case it is helpful to anyone else.

In the new implementation, an additional argument (`output`) has been added to the `todor()` function which defines the format of the results. Using this approach, the inline expression `r todor::todor(output = "text")` in an RMarkdown file now produces the following output in the rendered document (when applied to the `todor` package):

Screenshot 2021-05-12 130052

ToDos are grouped by filename, which is bolded for ease of reference.

At present, the `output` argument only accepts "text" (which returns markdown text) as a value. A possible further iternation could be "dataframe", which would allow users to asign the results of `todor` to a dataframe with the columns "file", "type" and "message".

Very happy to add extra functionality/documention/tests as part of the PR process - just let me know!

### CODE REVIEW

**Comment by dokato** [comment]
Fantastic! Thanks a lot @mcguinlu 👍  I was just going to answer below this issue, that I might find time on that over the weekend ;) I had a first look, looks great. I'll test that later today, or tomorrow and come back with comments if I have any.

File: R/todor.R
Comment (dokato) [inline R/todor.R:131]: in general, I don't mind purrr here, but looks like base `Map` will do the same, so loading whole package just for this functionality looks like overhead, could you call `Map(get, files)` instead (with new `get` name)?

File: R/todor.R
Comment (dokato) [inline R/todor.R:133]: for consistency with the rest of function you can drop `return` here

File: R/todor.R
Comment (dokato) [inline R/todor.R:115]: since we require a specific argument `marker`, this could be handled by ifelse and else could inform about available options is someone mistypes...

**Review by dokato — CHANGES_REQUESTED** [review]
well done @mcguinlu. This is a good start. I added a few comments. If you don't have time to address that, please add me push right to your clone, happy to help, cheers!

File: DESCRIPTION
Comment (mcguinlu) [reply DESCRIPTION:4]: Oh god, sorry - I used `usethis::use_package("PACKAGE_NAME")` to add the new dependencies, and it must have overwritten the Unicode when updating the DESCRIPTION.

Will fix!

File: R/todor.R
Comment (mcguinlu) [reply R/todor.R:124]: Yep, can do!

File: R/todor.R
Comment (mcguinlu) [reply R/todor.R:131]: Yeah, that makes a lot of sense! `purrr` is just my go-to - will swap it out for `Map`

File: R/todor.R
Comment (mcguinlu) [reply R/todor.R:115]: Yep - I had tried to have some error handling in the `IF NOT IN THEN STOP` function in the lines just before this, but can definitely improve it!

File: R/todor.R
Comment (mcguinlu) [reply R/todor.R:120]: My bad - got too excited to document it properly!

Will fix.

**Comment by mcguinlu** [comment]
Have left some comments on your requests - all very doable, and will have a look at implementing! 

One final query is that when I run `devtools::document()` on the package, it creates `.Rd` files for exported function in `man/` that are missing in your repo - is there a reason why you don't have a `todor_file.Rd` file in `man/` for example?

**Comment by dokato** [comment]
> One final query is that when I run `devtools::document()` on the package, it creates `.Rd` files for exported function in `man/` that are missing in your repo - is there a reason why you don't have a `todor_file.Rd` file in `man/` for example?

Great! I simply forgot to update Rd after major changes last year, I actually fixed that in #46 , so maybe to avoid merging conflicts you can skip it here. Once we merge in these two, I'll rebuild the docs separately.

File: DESCRIPTION
Comment (BastiHz) [reply DESCRIPTION:4]: This seems to be a bug in either usethis or the desc package. I'll look into it and file a bug report.

**Comment by mcguinlu** [comment]
Hi @dokato, 

All changes made now.

The only thing that might cause a flag for you is the use of `return()` when returning the list of markers. For some reason (you might know, but I don't) if I didn't use `return()` here, a NULL value was returned when using `output = "list"`.

Let me know if there are any further changes needed!

File: DESCRIPTION
Comment (mcguinlu) [reply DESCRIPTION]: Have removed `purrr` and fixed the diacritic.

File: R/todor.R
Comment (mcguinlu) [reply R/todor.R]: Have implemented this functionality now. Options are "list", "markdown" and "markers".

File: R/todor.R
Comment (mcguinlu) [reply R/todor.R]: Better docs have been added.

File: R/todor.R
Comment (mcguinlu) [reply R/todor.R]: Renamed and seperated out this function - now called `extract_markers_to_markdown()`

File: R/todor.R
Comment (mcguinlu) [reply R/todor.R]: `purrr::map()` now replaced with `Map()`

File: R/todor.R
Comment (mcguinlu) [reply R/todor.R]: Have improved error handling, which will give the following error if the `output` arg is mis-specified:

```
Output format not recognised. Available options are "markers", "list" and "markdown"
```

**Comment by dokato** [comment]
Fantastic, thanks @mcguinlu . I'll take care of the tests myself, though looks like a busy week ahead of me, but with some polishing touches over the weekend, I should bump up the version and update on CRAN soon.

> The only thing that might cause a flag for you is the use of `return()` when returning the list of markers. For some reason (you might know, but I don't) if I didn't use `return()` here, a NULL value was returned when using `output = "list"`.

Neglecting `return` only works on the variables that are at the very end of a function, so where you left it is as it should be.

**Review by dokato — APPROVED** [review]

**Pull request merged**
